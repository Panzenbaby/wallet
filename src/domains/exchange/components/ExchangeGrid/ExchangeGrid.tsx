import { EmptyBlock } from "app/components/EmptyBlock";
import React from "react";
import { useTranslation } from "react-i18next";

import { ExchangeCard } from "../ExchangeCard";

interface ExchangeGridProperties {
	exchanges: any[];
	onClick: any;
	onDelete: any;
	onOpenDetails: any;
}

export const ExchangeGrid = ({ exchanges, onClick, onDelete, onOpenDetails }: ExchangeGridProperties) => {
	const { t } = useTranslation();

	const actions = [
		{ label: t("COMMON.DETAILS"), value: "open-details" },
		{ label: t("COMMON.DELETE"), value: "delete" },
	];

	const handleExchangeAction = (exchange: any, action: any) => {
		switch (action?.value) {
			case "delete":
				onDelete(exchange);
				break;
			case "open-details":
				onOpenDetails(exchange);
				break;
		}
	};

	if (exchanges.length === 0) {
		return (
			<EmptyBlock data-testid="ExchangeGrid__empty-message">
				{t("EXCHANGE.PAGE_EXCHANGES.EMPTY_MESSAGE")}
			</EmptyBlock>
		);
	}

	return (
		<div data-testid="ExchangeGrid" className="grid grid-cols-3 w-full gap-4.5">
			{exchanges.map((exchange: any) => (
				<ExchangeCard
					key={exchange.id}
					actions={actions}
					exchange={exchange}
					onClick={() => onClick(exchange)}
					onSelect={(action: any) => handleExchangeAction(exchange, action)}
				/>
			))}
		</div>
	);
};
