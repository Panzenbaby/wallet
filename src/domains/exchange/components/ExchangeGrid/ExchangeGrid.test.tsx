import React from "react";
import { render, screen } from "utils/testing-library";

import { ExchangeGrid } from "./ExchangeGrid";

describe("ExchangeGrid", () => {
	it("should render", () => {
		const { asFragment } = render(<ExchangeGrid exchanges={[{ id: "exchange", updateStatus: {} }]} />);

		expect(asFragment()).toMatchSnapshot();
	});

	it("should render empty", () => {
		const { asFragment } = render(<ExchangeGrid exchanges={[]} />);

		expect(screen.getByTestId("ExchangeGrid__empty-message")).toBeInTheDocument();

		expect(asFragment()).toMatchSnapshot();
	});
});
