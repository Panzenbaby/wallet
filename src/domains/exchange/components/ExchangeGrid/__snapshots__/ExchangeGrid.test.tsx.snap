// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExchangeGrid should render 1`] = `
<DocumentFragment>
  <div
    class="grid grid-cols-3 w-full gap-4.5"
    data-testid="ExchangeGrid"
  >
    <div
      data-testid="PluginCard--exchange"
    >
      <div>
        <button
          class="Cardstyles__CardButton-sc-ajtauu-0 fyqAeZ"
          data-testid="Card"
          type="button"
        >
          <div
            class="flex items-center space-x-4"
          >
            <div
              class="overflow-hidden flex-shrink-0 rounded-lg w-25 h-25"
            >
              <div
                class="PluginImage__PluginImageWrapper-sc-10kztrf-0 fyVFIH bg-theme-primary-100 text-theme-primary-600 dark:bg-theme-secondary-800 dark:text-theme-secondary-700"
                data-testid="PluginImage"
              >
                <svg>
                  plugin-logo-placeholder.svg
                </svg>
              </div>
            </div>
            <div
              class="flex flex-col truncate"
            >
              <span
                class="text-sm font-semibold truncate text-theme-secondary-500 dark:text-theme-secondary-700"
              />
              <div
                class="flex items-center mt-1 space-x-2 text-lg font-bold truncate text-theme-primary-600 dark:text-theme-secondary-200"
              >
                <div
                  class="truncate"
                />
              </div>
            </div>
          </div>
          <div
            class="flex absolute -top-1 -right-1 items-center m-4 space-x-1"
          >
            <div
              class="flex items-center space-x-2"
            />
            <div
              class="relative"
            >
              <span
                data-testid="dropdown__toggle"
              >
                <div
                  class="flex overflow-hidden justify-center w-4"
                >
                  <div
                    class="sc-bdnxRM AFVmm p-1 transition-colors duration-200 cursor-pointer text-theme-primary-300 dark:text-theme-secondary-600 hover:text-theme-danger-400"
                    height="16"
                    width="16"
                  >
                    <svg>
                      settings.svg
                    </svg>
                  </div>
                </div>
              </span>
            </div>
          </div>
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`ExchangeGrid should render empty 1`] = `
<DocumentFragment>
  <div
    class="border border-solid border-theme-secondary-300 dark:border-theme-secondary-800 rounded-lg text-theme-secondary-text text-center leading-5 p-6"
    data-testid="ExchangeGrid__empty-message"
  >
    <div>
      There are currently no exchange providers available, please try again at a later time.
    </div>
  </div>
</DocumentFragment>
`;
